001-clarifications-eval.txt

```diff
@@ -0,0 +1,754 @@
+first paragraph:
+
+    --------------------------------------------------------------------
+    Each general operation represents a postfix operation on a simple
+    stack machine. Each element of the stack has a type and a value, and
+    can represent a value of any supported base type of the target
+    machine. Instead of a base type, elements can have a generic type,
+    which is an integral type that has the size of an address on the
+    target machine and unspecified signedness. The value on the top of
+    the stack after “executing” the DWARF expression is taken to be the
+    result (the address of the object, the value of the array bound, the
+    length of a dynamic string, the desired value itself, and so on).
+    --------------------------------------------------------------------
+
+... with the following:
+
+    --------------------------------------------------------------------
+    Each general operation represents a postfix operation on a simple
+    stack machine.
+
+    A value on the stack has a type and a literal value. It can
+    represent a literal value of any supported base type of the target
+    architecture. The base type specifies the size, encoding, and
+    endianity of the literal value.
+
+    There is a distinguished base type termed the generic type, which is
+    an integral type that has the size of an address in the target
+    architecture default address space, a target architecture defined
+    endianity, and unspecified signedness.
+
+    [non-normative] The generic type is the same as the unspecified type
+    used for stack operations defined in DWARF Version 4 and before.
+
+    An integral type is a base type that has an encoding of
+    DW_ATE_signed, DW_ATE_signed_char, DW_ATE_unsigned,
+    DW_ATE_unsigned_char, DW_ATE_boolean, or any target architecture
+    defined integral encoding in the inclusive range DW_ATE_lo_user to
+    DW_ATE_hi_user.
+    --------------------------------------------------------------------
+
+    > [For further discussion...]
+    > It may be desirable to add an implicit pointer base type encoding.
+    > It would be used for the type of the value that is produced when
+    > the DW_OP_deref* operation retrieves the full contents of an
+    > implicit pointer location storage created by the
+    > DW_OP_implicit_pointer operation. The literal value would record
+    > the debugging information entry and byte displacement specified by
+    > the associated DW_OP_implicit_pointer operation.
+    >
+    > It is unclear if DW_ATE_address is an integral type. GDB does not
+    > seem to consider it as integral.
+
+Insert the following after the end of the same section:
+
+    --------------------------------------------------------------------
+    Operations can act on entries on the stack, including adding entries
+    and removing entries.
+
+    Evaluation of an operation expression starts with an empty stack on
```

Comment:
What does 'operation' mean in 'operation expression'?

Reply:
See earlier comment that describes that a DWARF expression can either be a DWARF operation expression or a DWARF loclist expression. I suspect the text that defines those terms is in 004-locations-on-stack.txt .